### PR TITLE
[#452] Split auth dependency

### PR DESCRIPTION
**Pull request by paulodiniz**

### CODE REVIEW

File: lib/java/jwt/BUILD [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    "//:spring",
+    "//:springboot_security",
+    "//lib/java/log",
+    "//lib/java/spring/core:spring-core",
+    "@maven//:javax_servlet_javax_servlet_api",
+    "@maven//:javax_xml_bind_jaxb_api",
+]
+
+# Due to a bug in the abbreviation checkstyle rule
+# we need to skip style checking on this lib
```
Comment (chrismatix) [inline lib/java/jwt/BUILD:16]: Here too? :) Which abbreviation breaks?

File: lib/java/jwt/src/main/java/co/airy/jwt/Jwt.java [hunk 2]
```diff
@@ -20,7 +20,7 @@
 
 @Component
 public class Jwt {
-    private static final Logger log = AiryLoggerFactory.getLogger(Jwt.class);
+    private static final Logger log = AiryLoggerFactory.getLogger(co.airy.jwt.Jwt.class);
```
Comment (chrismatix) [inline lib/java/jwt/src/main/java/co/airy/jwt/Jwt.java:23]: This can stay `Jwt.class` I think

File: backend/api/communication/BUILD [hunk 3]
```diff
@@ -11,6 +11,7 @@ app_deps = [
     "//backend:channel",
     "//backend:metadata",
     "//backend:read-receipt",
+    "//lib/java/jwt:airy-jwt",
```
Comment (chrismatix) [inline backend/api/communication/BUILD:14]: If you make `spring-auth` export `//lib/java/jwt:airy-jwt` (usually we avoid this, but here it seems ok) you only have to change the dependencies of the test library 🙏

Comment (paulodiniz) [reply backend/api/communication/BUILD:14]: NICE 👍  TIL

File: lib/java/spring/jwt/BUILD [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    "//:lombok",
+    "//:spring",
+    "//:springboot_security",
+    "//lib/java/log",
+    "//lib/java/spring/core:spring-core",
+    "@maven//:javax_servlet_javax_servlet_api",
+    "@maven//:javax_xml_bind_jaxb_api",
+]
+
+java_library(
```
Comment (chrismatix) [inline lib/java/spring/jwt/BUILD:15]: This should use our own `custom_java_library`:
```
load("//tools/build:java_library.bzl", "custom_java_library")
```

**Review by chrismatix — APPROVED** [review]

**Pull request merged**